flo-freezer/src/test/java/com/spotify/flo/freezer/PersistingContextTest.java

```diff
@@ -0,0 +1,81 @@
+/*-
+ * -\-\-
+ * Flo Freezer
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.flo.freezer;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.nio.file.Path;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class PersistingContextTest {
+
+  @Rule public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test
+  public void exceptionSerialization() throws Exception {
+    final RuntimeException bar = new RuntimeException("bar");
+    bar.addSuppressed(new IOException("baz", new InterruptedException("quux")));
+    final Exception exception1 = new Exception("foo", bar);
+    exception1.addSuppressed(new FoobarException());
+    final Exception exception = exception1;
+
+    final Path exceptionFile = temporaryFolder.getRoot().toPath().resolve("exception");
+    PersistingContext.serialize(exception, exceptionFile);
+
+    final Exception deserialized = PersistingContext.deserialize(exceptionFile);
+
+    // Verify that stack trace can be accessed and printed and seems to be correct
+
+    deserialized.printStackTrace();
```

Comment:
Is this needed?

Reply:
Yes, like the comment says, we explicitly want to verify this.

The problem here was that exceptions that were deserialized would throw NPE when `printStackTrace()` was called on them

Reply:
But the lines below uses print writer to verify each line explicitly.

Reply:
Will remove the loop below.